### PR TITLE
MS5837 pressure sensor

### PR DESCRIPTION
**Pull request by vanwinkeljan**
Added sensor driver for MS5837 pressure sensor

### CODE REVIEW

File: samples/sensor/ms5837/README.rst [hunk 1]
```diff
@@ -0,0 +1,52 @@
+.. _ms5837-sample:
+
+MS5837 Sensor Sample
+####################
+
+Overview
+********
+
+Sample application that every 10 seconds retrieves the pressure and temperature
```
Comment (dbkinder) [inline samples/sensor/ms5837/README.rst]: This should be a complete sentence, so how about:

    This sample application retrieves the pressure and temperature
    from a MS5837 sensor every 10 seconds, and prints this
    information to the UART console.

**Review by dbkinder — CHANGES_REQUESTED** [review]
Small change to the intro, otherwise looks good. Thanks!

**Review by dbkinder — APPROVED** [review]
+1 for docs, thanks.

**Comment by vanwinkeljan** [comment]
DTS bindings are missing, I will add them

File: samples/sensor/ms5837/src/main.c [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+#include <stdio.h>
+#include <zephyr.h>
+
+#define SYS_LOG_DOMAIN "main"
+#define SYS_LOG_LEVEL SYS_LOG_LEVEL_DEBUG
+#include <logging/sys_log.h>
+
+void main(void)
+{
+	struct device *dev = device_get_binding("MS5837");
```
Comment (MaureenHelm) [inline samples/sensor/ms5837/src/main.c]: Don't hardcode the string, get the label from device tree instead.

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: drivers/sensor/ms5837/ms5837.c [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+
+static int ms5837_get_measurement(struct device *i2c_master, u32_t *val,
+				  u8_t cmd, u8_t delay)
+{
+	u8_t adc_read_cmd = MS5837_CMD_CONV_READ_ADC;
+	int err;
+
+	*val = 0;
+
+	err = i2c_write(i2c_master, &cmd, 1, MS5837_ADDR);
```
[the remark this answers is not in the document]
Comment (vanwinkeljan) [reply drivers/sensor/ms5837/ms5837.c]: To trigger a conversion on the MS5837 only a single byte command should be issued and as such i2c_reg_write_byte can not be used as it transfers 2 bytes (reg address + value).

File: drivers/sensor/ms5837/ms5837.c [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+	data->i2c_master =
+	    device_get_binding(CONFIG_MS5837_I2C_MASTER_DEV_NAME);
+	if (data->i2c_master == NULL) {
+		SYS_LOG_DBG("i2c master %s not found",
+			    CONFIG_MS5837_I2C_MASTER_DEV_NAME);
+		return -EINVAL;
+	}
+
+	cmd = MS5837_CMD_RESET;
+	err = i2c_write(data->i2c_master, &cmd, 1, MS5837_ADDR);
```
[the remark this answers is not in the document]
Comment (vanwinkeljan) [reply drivers/sensor/ms5837/ms5837.c]: Reset is a single byte command i2c_reg_write_byte is a 2 byte transfer (reg + value)

[the remark this answers is not in the document]
Comment (MaureenHelm) [reply drivers/sensor/ms5837/ms5837.c]: Ok

File: drivers/sensor/ms5837/ms5837.c [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+	int err;
+	u8_t cmd;
+
+	data->pressure = 0;
+	data->temperature = 0;
+
+	data->presure_conv_cmd = MS5837_CMD_CONV_P_256;
+	data->presure_conv_delay = 1;
+	data->temperature_conv_cmd = MS5837_CMD_CONV_T_256;
+	data->temperature_conv_delay = 1;
```
[the remark this answers is not in the document]
Comment (vanwinkeljan) [reply drivers/sensor/ms5837/ms5837.c]: Added set attribute function to the driver as these are configurable parameters

File: drivers/sensor/ms5837/ms5837.c [hunk 6]
```diff
@@ -0,0 +1,310 @@
...
+	if (err < 0) {
+		return err;
+	}
+
+	return 0;
+}
+
+static struct ms5837_data ms5837_data;
+
+const struct ms5837_config ms5837_config = {
```
Comment (MaureenHelm) [inline drivers/sensor/ms5837/ms5837.c]: This should be static too

File: drivers/sensor/ms5837/ms5837.c [hunk 7]
```diff
@@ -0,0 +1,310 @@
...
+	if (err < 0) {
+		return err;
+	}
+
+	*val = sys_be16_to_cpu(*val);
+
+	return 0;
+}
+
+int ms5837_init(struct device *dev)
```
Comment (MaureenHelm) [inline drivers/sensor/ms5837/ms5837.c]: Make this static

File: dts/bindings/sensor/te,ms5837.yaml [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+description: >
+    This binding gives a base representation of the MS5837 digital pressure
+    sensor
+
+inherits:
+    !include i2c-device.yaml
+
+properties:
+    compatible:
+      constraint: "te,ms5837"
```
Comment (MaureenHelm) [inline dts/bindings/sensor/te,ms5837.yaml]: Let's make this `meas,ms5837` to be consistent with the linux kernel

File: samples/sensor/ms5837/dts_fixup.h [hunk 9]
```diff
@@ -0,0 +1,23 @@
+#if defined(CONFIG_HAS_DTS_I2C)
```
Comment (MaureenHelm) [inline samples/sensor/ms5837/dts_fixup.h]: Add a license/copyright header

File: samples/sensor/ms5837/nrf52840_pca10056.overlay [hunk 10]
```diff
@@ -0,0 +1,9 @@
+&i2c1 {
```
Comment (MaureenHelm) [inline samples/sensor/ms5837/nrf52840_pca10056.overlay]: Add a license/copyright header

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: drivers/sensor/ms5837/ms5837.c [hunk 11]
```diff
@@ -0,0 +1,310 @@
...
+	s64_t dT;
+	s64_t OFF;
+	s64_t SENS;
+	s64_t temp_sq;
+	s32_t Ti;
+	s32_t OFFi;
+	s32_t SENSi;
+
+	/* first order compensation as per datasheet
+	 * (https://tinyurl.com/y9eu4c9t) section PRESSURE AND TEMPERATURE
```
[the remark this answers is not in the document]
Comment (vanwinkeljan) [reply drivers/sensor/ms5837/ms5837.c]: add url to landing page for ms5837 as url is of reasonable length

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**